### PR TITLE
python3Packages.pydateinfer: init at 0.3.0

### PR DESCRIPTION
**Pull request by mbalatsko**
## Description of changes
Adding one of the requirements needed for https://github.com/NixOS/nixpkgs/issues/247089 . I want to separate `mindsdb` requirements initialization into separate PRs bacause adding it to `nixpkgs` requires more than 15 new packages. I believe this way it would be easier to review.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mbalatsko** [comment]
Result of `nixpkgs-review pr 255506` on x86_64-linux:
```
2 packages built:
python310Packages.pydateinfer python311Packages.pydateinfer
```

**Comment by mbalatsko** [comment]
@GrahamcOfBorg build python310Packages.pydateinfer python311Packages.pydateinfer

File: pkgs/development/python-modules/pydateinfer/default.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  src = fetchFromGitHub {
+    owner = "wdm0006";
+    repo = "dateinfer";
+    rev = "0.3.0,"; # yes the comma is required, this is correct name of git tag
+    hash = "sha256-0gy7wfT/uMTmpdIF2OPGVeUh+4yqJSI2Ebif0Lf/DLM=";
+  };
+
+  patches = [
+    ./fix_examples_path_in_tests.patch
+  ];
```
Comment (natsukium) [inline pkgs/development/python-modules/pydateinfer/default.nix:27]: Can we replace it with
```
preCheck = ''
  cd dateinfer
'';
```
?

Comment (mbalatsko) [reply pkgs/development/python-modules/pydateinfer/default.nix]: fixed

**Comment by mbalatsko** [comment]
@GrahamcOfBorg build python310Packages.pydateinfer python311Packages.pydateinfer

File: pkgs/development/python-modules/pydateinfer/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+buildPythonPackage rec {
+  pname = "pydateinfer";
+  version = "0.3.0";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "wdm0006";
+    repo = "dateinfer";
+    rev = "0.3.0,"; # yes the comma is required, this is correct name of git tag
```
Comment (happysalada) [inline pkgs/development/python-modules/pydateinfer/default.nix:18]:
```suggestion
    rev = "${version},"; # yes the comma is required, this is correct name of git tag
```

Comment (happysalada) [reply pkgs/development/python-modules/pydateinfer/default.nix:18]: This should make it easier to update

File: pkgs/development/python-modules/pydateinfer/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    unittestCheckHook
+    pyyaml
+    argparse
+  ];
+  pythonImportsCheck = [ "dateinfer" ];
+
+  meta = with lib; {
+    description = "Infers date format from examples";
+    homepage = "https://pypi.org/project/pydateinfer/";
+    license = licenses.asl20;
```
Comment (happysalada) [inline pkgs/development/python-modules/pydateinfer/default.nix:37]: Is there a changelog in the source repo by any chance ?

**Review by happysalada — COMMENTED** [review]
appart from two comments this generally looks good.

File: pkgs/development/python-modules/pydateinfer/default.nix (hunk 3 above)
Comment (mbalatsko) [reply pkgs/development/python-modules/pydateinfer/default.nix:37]: There is no changelog file and no realeses

Comment (mbalatsko) [reply pkgs/development/python-modules/pydateinfer/default.nix:37]: We can theoretically add it as https://github.com/wdm0006/dateinfer/compare/v0.2.0..0.3.0%2C but I'm not sure that it is the right way to define changelog

File: pkgs/development/python-modules/pydateinfer/default.nix (hunk 2 above)
Comment (mbalatsko) [reply pkgs/development/python-modules/pydateinfer/default.nix]: Fixed

**Comment by happysalada** [comment]
@mbalatsko I missed this, but the patch is now not used anymore right ? can you remove it?

**Comment by mbalatsko** [comment]
@happysalada oops my bad, good catch! Deleted!

**Pull request merged**